src/components/UserImpact.js

```diff
@@ -0,0 +1,142 @@
+  }
+  const handleRewardDialogClose = () => setRewardDialogOpen(false)
+
+  const handleClaimReferralNotification = async () => {
+    setReferralRewardDialogOpen(true)
+  }
+  const handleReferralRewardDialogClose = () => {
+    setReferralRewardDialogOpen(false)
+    UpdateImpactMutation(userId, CAT_CHARITY, {
+      claimPendingUserReferralImpact: true,
+    })
+  }
+  const classes = useStyles()
+  return (
+    <div>
+      <ImpactCounter
+        includeNumber
+        className={classes.impactCounter}
+        number={userImpactMetric}
+        progress={(1 - visitsUntilNextImpact / 14) * 100}
+      />
+      <ImpactDialog
+        modalType="confirmImpact"
+        open={confirmDialogOpen}
+        buttonOnClick={handleConfirmDialogClose}
+      />
+      <ImpactDialog
+        modalType="impactWalkMe"
+        open={alertDialogOpen}
+        onClose={handleAlertDialogClose}
+      />
+      <ImpactDialog
+        modalType="claimImpactReward"
+        open={rewardDialogOpen}
+        buttonOnClick={handleRewardDialogClose}
+        user={user}
+      />
+      <ImpactDialog
+        modalType="claimReferralReward"
+        open={referralRewardDialogOpen}
+        buttonOnClick={handleReferralRewardDialogClose}
+        referralImpact={pendingUserReferralImpact}
+        user={user}
+      />
+      {showReward && (
+        <Notification
+          text={
+            <Typography>
+              You did it! You just turned your tab into a treat for a cat. Keep
+              it up, and do good with every new tab!"
+            </Typography>
+          }
+          buttonText="Hooray"
+          buttonOnClick={handleClaimReward}
+        />
+      )}
+      {referralRewardNotificationOpen && (
+        <Notification
+          text={
+            <Typography>
+              congrats! You recruited{' '}
+              <span style={{ fontWeight: 'bold' }}>
+                {`${pendingUserReferralCount} friend${
+                  pendingUserReferralCount > 1 ? 's' : ''
+                } `}
+              </span>
+              to help shelter cats just by opening tabs. To celebrate, we'll
+              give a treat to an extra {pendingUserReferralImpact} cats.
+            </Typography>
+          }
+          buttonText="Claim"
+          buttonOnClick={handleClaimReferralNotification}
+        />
+      )}
+    </div>
+  )
+}
+
+UserImpact.displayName = 'UserImpact'
+UserImpact.propTypes = {
+  userImpact: PropTypes.shape({
+    visitsUntilNextImpact: PropTypes.number.isRequired,
+    pendingUserReferralImpact: PropTypes.number.isRequired,
+    pendingUserReferralCount: PropTypes.number.isRequired,
+    userImpactMetric: PropTypes.number.isRequired,
+    confirmedImpact: PropTypes.bool.isRequired,
+    hasClaimedLatestReward: PropTypes.bool.isRequired,
+  }).isRequired,
+  userId: PropTypes.string,
+  user: PropTypes.shape({
+    username: PropTypes.string,
+  }),
+}
+
+UserImpact.defaultProps = {
+  user: {},
+  userId: '',
```

Comment:
Use `undefined` values here (or remove defaults altogether if you make the props required).